### PR TITLE
feat: added promisesAllSettled method that resolves after all of the given promises have either fulfilled or rejected 

### PR DESCRIPTION
**Pull request by kuldeepk-dazn**
## What did you implement:

Closes #ROKU-142

Added promisesAllSettled function that takes an array of promises as an input and returns a promise that resolves after all of the given promises have either fulfilled or rejected with an array of objects that each describes the outcome of each promise.

## How did you implement it:

Function loops through the given promises and wait until all promises have been resolved or rejected. Once all promises are settled it returns the promise with an array of objects [status, value] for all the promises.

## How can we verify it:

/src/components/promise/_tests/promisesAllSettled.test.brs

## Todos:

- [ ] Write documentation (if required)
- [ ] Fix linting errors
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: src/components/promise/promisesAllSettled.brs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  for each _promise in context.promises
+    if (_promise.status = _promise.STATUS_FULFILLED OR _promise.status = _promise.STATUS_REJECTED)
+      context.onPromiseSettled(_promise.value, context)
+    else
+      _promise.then(context.onPromiseSettled, context.onPromiseSettled, context)
+    end if
+  end for
+
+  return context.promise
+end function
```
Comment (pawelhertman) [inline src/components/promise/promisesAllSettled.brs:41]: empty line is missing

**Review by pawelhertman — APPROVED** [review]
Great job 👏 Just one small remark. Approved in advance

File: src/components/promise/promisesAllSettled.brs (hunk 1 above)
Comment (kuldeepk-dazn) [reply src/components/promise/promisesAllSettled.brs]: Added.

File: src/components/promise/promisesAllSettled.brs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      m.promise.resolve(promisesValues)
+      m.promises.clear()
+    end if
+  end sub
+
+  for each _promise in context.promises
+    if (_promise.status = _promise.STATUS_FULFILLED OR _promise.status = _promise.STATUS_REJECTED)
+      context.onPromiseSettled(_promise.value, context)
+    else
+      _promise.then(context.onPromiseSettled, context.onPromiseSettled, context)
```
Comment (bchelkowski) [inline src/components/promise/promisesAllSettled.brs:36]:
```suggestion
      _promise.finally(context.onPromiseSettled, context)
```
would just change this small thing

**Review by bchelkowski — APPROVED** [review]

**Pull request merged**